Sources/NIO/Bootstrap.swift

```diff
@@ -325,6 +325,47 @@ private extension Channel {
     }
 }
 
+/// `ClientTransportBootstrap` is implemented by various underlying transport mechanisms. Typically,
+/// this will be the BSD Sockets API implemented by `ClientBootstrap`.
+public protocol ClientTransportBootstrap {
```

Comment:
This would need to be called `NIOClientTransportBootstrap` to meet our API naming guidelines.